### PR TITLE
Fix #5230: QueryRule equals + integration test method for caching performance measurement

### LINKED ISSUE #5230: QueryRule equals method too strict
#### Reproduce
Compare two identical QueryRule instances, but with Entity value set to two instances of an entity with the same idValue.

#### Expected
equals() returns true

#### Observed
equals() returns false

This causes unnecessary misses in the L3 / query cache.

### PR DESCRIPTION
**Pull request by Mark-de-Haan**
#### Checklist
- [x] Unit tests

### CODE REVIEW

**Comment by bartcharbon** [comment]
please fix the failing test

File: molgenis-data-cache/src/main/java/org/molgenis/data/cache/l3/L3CacheRepositoryDecorator.java [hunk 1]
```diff
@@ -53,11 +57,19 @@ public L3CacheRepositoryDecorator(Repository<Entity> decoratedRepository, L3Cach
 	@Override
 	public Stream<Entity> findAll(Query<Entity> query)
 	{
-		if (transactionInformation.isRepositoryCompletelyClean(getName()) && cacheable && query.getPageSize() > 0
-				&& query.getPageSize() <= MAX_PAGE_SIZE)
+		if (transactionInformation.isRepositoryCompletelyClean(getName()))
+		{
+			// FIXME page size for metadata is always 0, and batching is done by the postgres repository
```
Comment (bartcharbon) [inline molgenis-data-cache/src/main/java/org/molgenis/data/cache/l3/L3CacheRepositoryDecorator.java]: is there a story or issue to make sure these FIXME's get addressed?

Comment (fdlk) [inline molgenis-data-cache/src/main/java/org/molgenis/data/cache/l3/L3CacheRepositoryDecorator.java]: I've already added it under the header "Issues" in the FD/TD. Can also file a bug here.

**Comment by fdlk** [comment]
Closed for #5239

**Pull request closed without merging**